transaction/core/src/membership_proofs/mod.rs

```diff
@@ -95,7 +95,7 @@ pub fn compose_adjacent_membership_elements(
     // this test, then one of the ranges must be reversed, contrary to the
     // precondition.
     if !bool::from(a_is_left ^ b_is_left) {
-        return Err(());
+        return Err(Error::PreconditionFailed);
```

Comment:
I think instead of making a new variant, we should reconcile this with line 184 below, which maps `()` onto the "final" error type?

Reply:
That would force us to pass context to this function just for the purpose of returning an error. That doesn't seem to make sense to me.